core/src/main/java/io/grpc/DnsNameResolverFactory.java

```diff
@@ -63,7 +65,7 @@ public NameResolver newNameResolver(URI targetUri, Attributes params) {
       Preconditions.checkArgument(targetPath.startsWith("/"),
           "the path component (%s) of the target (%s) must start with '/'", targetPath, targetUri);
       String name = targetPath.substring(1);
-      return new DnsNameResolver(targetUri.getAuthority(), name, params);
+      return new DnsNameResolver(targetUri.getAuthority(), name, params, GrpcUtil.TIMER_SERVICE);
```

Comment:
If we are using this shared single-threaded `ScheduledExecutorService` we should probably make sure that we don't do blocking DNS resolution on that executor thread. Instead, we may need to use a combination of `SHARED_CHANNEL_EXECUTOR` and the `TIMER_SERVICE`.